torch/distributed/elastic/utils/store.py

```diff
@@ -93,6 +101,46 @@ def synchronize(
         return agent_data
 
 
+def _try_detecting_missing_ranks(
+    store,
+    world_size: int,
+    key_prefix: str,
+    rank: int,
+    rank_decoder: Callable[[int], str],
+    trace_timeout: float,
+) -> Optional[Iterable[str]]:
+    store.set(f"{key_prefix}{rank}{_TRACE}", "<val_ignored>")
+
+    missing_rank_info = set()
+    ranks_missing = 0
+    if rank == 0:
```

Comment:
One thought since this is a more general TCPStore implementation - for small scale jobs, which would be the majority in PyTorch, it's fine for all ranks to report this info. I think that should probably the default. And if the user opts to have this done by a single rank (which could be an arg to the barrier), then the user can pass in the rank or ranks to report this info.

Reply:
Bit torn here. We could do this on all ranks but at the cost of extra operations that will be over the wire. This approach reduces number operations that go over the wire.